# Titanic Using SVM - Kaggle Competition

Author: Szymon Pawłowski\
Date: 10.01.2021r.

# Kaggle Titanic Competition
It is one of the first challenges every ML beginner should dive in. \
In this competition the main goal is to predict which passengers survived the Titanic shipwreck \
using given data and creating a ML model. Here the SVM is used for prediction. 

# The data
The data has been split into two groups: training set and test set. Columns they contain:
* Survival - did passenger survive - 0=no, 1=yes
* Pclass - ticket class - 1=1st, 2=2nd, 3=3rd
* Name 
* Sex - sex - m/f 
* Age - age in years
* Sibsp - number of siblings / spouses aboard the Titanic 
* Parch - number of parents / children aboard the Titanic 
* Ticket - ticket number 
* Fare - passenger fare
* Cabin - cabin number 
* Embarked - port of embarkation - C=Cherbourg, Q=Quennstown, S=Southampton

# Libraries

In [5]:
import pandas as pd
import numpy as np
from matplotlib import pyplot as plt
import matplotlib.ticker as mticker
import seaborn as sns
%matplotlib inline
import warnings
warnings.filterwarnings('ignore')

# Loading data and first sight into it

In [14]:
#Loading data
test = pd.read_csv("test.csv")
train = pd.read_csv("train.csv")

In [15]:
#Combining into one data frame
df = pd.concat([train,test])

In [24]:
df.head()

PassengerId  Survived  Pclass  \
0            1       0.0       3   
1            2       1.0       1   
2            3       1.0       3   
3            4       1.0       1   
4            5       0.0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [25]:
df.tail()

PassengerId  Survived  Pclass                          Name     Sex  \
413         1305       NaN       3            Spector, Mr. Woolf    male   
414         1306       NaN       1  Oliva y Ocana, Dona. Fermina  female   
415         1307       NaN       3  Saether, Mr. Simon Sivertsen    male   
416         1308       NaN       3           Ware, Mr. Frederick    male   
417         1309       NaN       3      Peter, Master. Michael J    male   

      Age  SibSp  Parch              Ticket      Fare Cabin Embarked  
413   NaN      0      0           A.5. 3236    8.0500   NaN        S  
414  39.0      0      0            PC 17758  108.9000  C105        C  
415  38.5      0      0  SOTON/O.Q. 3101262    7.2500   NaN        S  
416   NaN      0      0              359309    8.0500   NaN        S  
417   NaN      1      1                2668   22.3583   NaN        C

In [26]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1309 entries, 0 to 417
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  1309 non-null   int64  
 1   Survived     891 non-null    float64
 2   Pclass       1309 non-null   int64  
 3   Name         1309 non-null   object 
 4   Sex          1309 non-null   object 
 5   Age          1046 non-null   float64
 6   SibSp        1309 non-null   int64  
 7   Parch        1309 non-null   int64  
 8   Ticket       1309 non-null   object 
 9   Fare         1308 non-null   float64
 10  Cabin        295 non-null    object 
 11  Embarked     1307 non-null   object 
dtypes: float64(3), int64(4), object(5)
memory usage: 132.9+ KB


We can see that Cabin has only 295 non-null observations from 1309. It is too hard to handle those NA's so we will have remove this predictor (either way it was also hard to get information from it). What is more, we need to change object types into binary classification. From Name we can extract special titles which can improve our model. What is more, from SibSp and Parch we can get information about family size and categorize it. Let's see what our data say to us.

# Exploratory Data Analysis (EDA)